### DISCUSSION TITLE
how do I check a channel connection status

### DISCUSSION THREAD

**Discussion by azdave2020** [nextgenhealthcare/connect#5014] [Q&A]
Hi all.
In Mirth 3.10  or higher I would like to check a channels connection for a status if Idle and if Idle for more than X period of time, send a message.  I looked through the ChannelUtil as well as some other docs on the interwebs and have not really found a solution. 

Any guidance is appreciated.
Thanks
AZDave

**Comment by pacmano1** [comment]
In your case does idle = "has not processed message" in some period of time?

**Reply by azdave2020** [reply]
Hi pacmano
In my case it literally means idle, but you are right it could me no messages in a certain period. the risk with "no messages" is that we can go 8 hours without a message and if the vendor is AOK the channel stays connected. When it toggles to Idle we have to have the vendor reboot.
Dave
![image](https://user-images.githubusercontent.com/90641786/153252013-213fdd1c-5c1e-4a58-a74f-a2520e8f9a57.png)

**Comment by jonbartels** [comment]
Can you see the condition you want to detect in the "connections" tab in the main dashboard?

If you can - everything in the dashboard is driven by an API call to MC and you can just make those API calls either using HTTPS or using the internal classes.

**Reply by azdave2020** [reply]
Hi Jon connections is a column in the dashboard.  So far I have not been able to find any MC api or class that will let me return that status of the column.  Any suggestions?

**Reply by jonbartels** [reply]
Take a look at `com.mirth.connect.plugins.dashboardstatus.DashboardConnectorStatusServletInterface` at https://github.com/nextgenhealthcare/connect/blob/e66d1122dbeebfcf5a8939beb93d720b2d5071fb/server/src/com/mirth/connect/plugins/dashboardstatus/DashboardConnectorStatusServletInterface.java . It's actually built as an extension so its API may not be included in Swagger. 

This is what populates the "Connection" column in the dashboard and the "Connection Log" tab in the dashboard.

**Reply by azdave2020** [reply]
Thanks that got me in the right direction!
